crates/ln-dlc-node/src/node.rs

```diff
@@ -679,9 +681,107 @@ impl Node {
             .insert(intercept_scid, target_node);
         intercept_scid
     }
+
+    pub async fn initiate_dlc_channel(
+        &self,
+        channel_details: &ChannelDetails,
+        contract_input: &ContractInput,
+    ) -> Result<()> {
+        let announcement = tokio::task::spawn_blocking({
+            let oracle = self.oracle.clone();
+            let event_id = contract_input.contract_infos[0].oracles.event_id.clone();
+            move || {
+                oracle
+                    .get_announcement(&event_id)
+                    .map_err(|e| anyhow!(e.to_string()))
+            }
+        })
+        .await??;
+
+        let sub_channel_offer = self
+            .sub_channel_manager
+            .offer_sub_channel(
+                &channel_details.channel_id,
+                contract_input,
+                &[vec![announcement]],
+            )
+            .unwrap();
+
+        self.dlc_message_handler.send_subchannel_message(
+            channel_details.counterparty.node_id,
+            SubChannelMessage::Request(sub_channel_offer),
+        );
+
+        Ok(())
+    }
+
+    pub fn initiate_accept_dlc_channel_offer(&self, channel_id: &[u8; 32]) -> Result<()> {
+        let channel_id_hex = hex::encode(channel_id);
+
+        tracing::info!(channel_id = %channel_id_hex, "Accepting DLC channel offer");
+
+        let (node_id, accept_sub_channel) = self
+            .sub_channel_manager
+            .accept_sub_channel(channel_id)
+            .map_err(|e| anyhow!(e.to_string()))?;
+
+        self.dlc_message_handler
+            .send_subchannel_message(node_id, SubChannelMessage::Accept(accept_sub_channel));
+
+        Ok(())
+    }
+
+    pub fn process_incoming_messages(&self) -> Result<()> {
```

Comment:
Does this have to be a manual step?

Reply:
i fear so yes. have a look at our discussion with tobi. 

https://matrix.to/#/!YOUZSBmuJuaGzmxzaA:matrix.org/$AEqhGweYmktqLYTCUBR8ZT13coxfOJzcpYCd-pFe2MU?via=matrix.org

Reply:
> ... tobi 

--> Tibo ;)